File: rb/features.rb
# RealtimeWeather SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module RealtimeWeatherFeatures
  def self.make_feature(name)
    case name
    when "base"
      RealtimeWeatherBaseFeature.new
    when "test"
      RealtimeWeatherTestFeature.new
    else
      RealtimeWeatherBaseFeature.new
    end
  end
end

File: py/features.py
# RealtimeWeather SDK feature factory

from feature.base_feature import RealtimeWeatherBaseFeature
from feature.test_feature import RealtimeWeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RealtimeWeatherBaseFeature(),
        "test": lambda: RealtimeWeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
